notmuch-addrlookup.c

```diff
@@ -96,24 +96,31 @@ load_notmuch_settings (void)
   gchar *config_path = NULL;
   gboolean success = FALSE;
 
-  if (g_getenv ("NOTMUCH_CONFIG"))
+  if(notmuch_config_path_lookupvar)
+    config_path = notmuch_config_path_lookupvar;
+  else if (g_getenv ("NOTMUCH_CONFIG"))
     config_path = g_strdup (g_getenv ("NOTMUCH_CONFIG"));
-  else if(!notmuch_config_path_lookupvar)
-    config_path = g_strdup_printf ("%s/.notmuch-config", g_get_home_dir ());
+  else if (g_getenv ("NOTMUCH_PROFILE")){
+    if (g_getenv("XDG_CONFIG_HOME")){
```

Comment:
You should use [g_get_user_config_dir()](https://docs.gtk.org/glib/func.get_user_config_dir.html) which will work even if the `XDG_CONFIG_HOME` environment variable is undefined. Then you don't need to check if `g_getenv()` returned a non-`NULL` value.

To check whether the file exists you can use [g_file_new_build_filename()](https://docs.gtk.org/gio/type_func.File.new_build_filename.html) to calculate the path and create a `GFile` for it at the same time, then use [g_file_query_exists()](https://docs.gtk.org/gio/method.File.query_exists.html). Finally, if the file is present, its full path can be obtained with [g_file_get_path()](https://docs.gtk.org/gio/method.File.get_path.html).